presto-hive/src/main/java/io/prestosql/plugin/hive/HiveTableHandle.java

```diff
@@ -153,6 +157,17 @@ public HiveTableHandle withAnalyzeColumnNames(Set<String> analyzeColumnNames)
                 Optional.of(analyzeColumnNames));
     }
 
+    @Override
+    public void validateScan(ConnectorSession session)
+    {
+        if (HiveSessionProperties.isQueryPartitionFilterRequired(session) && !partitionColumns.isEmpty() && getEnforcedConstraint().isAll()) {
+            String partitionColumnNames = partitionColumns.stream().map(n -> n.getName()).collect(Collectors.joining(","));
```

Comment:
Will it allow queries with predicate expression on partition column?

Reply:
Can you give me an example? If the query ends up with an enforced constraint, it'll be allowed.

Reply:
Suppose there is a table 'T' with a partitioned column 'C' of type 'varchar' and you fired
```select * from T where CAST(C as int) = 10```, then I think this will throw an exception but it shouldn't, right?

Reply:
Yes, that'd be flagged as an error (assuming the partition filter enforcement feature is enabled). I suspect that query would execute more efficiently written as `select * from T where C = '10'` which is the kind of thing this feature is supposed to catch.

Reply:
Partition pruning happens in both cases, in case of `C = '10'` the predicate can be pushed down hive metastore as well to improve planning time. In case of `CAST(C as int) = 10` Presto will still run an expression interpreter to prune partitions after the hive metastore calls. So I think we'd want to allow such cases since partition pruning is performed and some extra CASTs or date/time conversion functions are often present in predicates on partition columns in queries generated by BI tools.

Reply:
Makes sense - thanks for the suggestion. I've pushed a commit so that this case passes the partition filter check.